airbyte-cdk/bulk/toolkits/extract-cdc/src/main/kotlin/io.airbyte.cdk/read/CdcGlobalLockResource.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.read
+
+import io.micronaut.context.annotation.Replaces
+import java.util.concurrent.atomic.AtomicBoolean
+
+@Replaces(GlobalLockResource::class)
```

Comment:
Not familiar with `@Replaces` but I guess we need that to only apply to GlobalLockResource when running in the context of CDC.
i.e is it going to affect the GlobalLockResource in jdbc?

Reply:
Trying to run integration tests I'm getting 
```
Message: No bean of type [io.airbyte.cdk.read.CdcGlobalLockResource] exists
```

Reply:
```suggestion
@Singleton
@Replaces(GlobalLockResource::class)
```

Reply:
as you could tell, I didn't actually test this (had to catch the train)